scopes/harmony/aspect-loader/plugins.ts

```diff
@@ -116,17 +121,48 @@ export class Plugins {
   /**
    * Get the plugin files from the component.
    */
-  static files(component: Component, defs: PluginDefinition[], resolvePath?: (path: string) => string): string[] {
+
+  static files(
+    component: Component,
+    defs: PluginDefinition[],
+    logger: Logger,
+    resolvePath?: (path: string) => string
+  ): string[] {
     const files = defs.flatMap((pluginDef) => {
-      const matches =
-        typeof pluginDef.pattern === 'string'
-          ? component.filesystem.byGlob([pluginDef.pattern])
-          : component.filesystem.byRegex(pluginDef.pattern);
+      const matches = this.getFileMatches(component, pluginDef);
+
+      const warningMessage = this.constructWarningMessage(component);
+      if (matches.length === 0 && warningMessage && !Plugins.checkedComponents.has(component.id.toString())) {
+        logger.consoleWarning(warningMessage);
+        Plugins.checkedComponents.add(component.id.toString());
+      }
 
       return matches.map((file) => {
         return resolvePath ? resolvePath(file.relative) : file.path;
       });
     });
     return files;
   }
+
+  private static getFileMatches(component: Component, pluginDef: PluginDefinition): any[] {
+    return typeof pluginDef.pattern === 'string'
+      ? component.filesystem.byGlob([pluginDef.pattern])
+      : component.filesystem.byRegex(pluginDef.pattern);
+  }
+
+  private static isImportedComponent(component: Component): boolean {
```

Comment:
Why is it important that it's imported it not?
And how this actually check if it is imported?
We shouldn't have any diff between imoorted or authored components.

Reply:
If the files are absolute path, then it is imported
The thing is that sometimes it would not load an installed env for some reason at first attempt. Not sure why. I will try to look into it.

Reply:
ok now I understand
If we have an env installed in the workspace but not set to any component then it will fail because it will not find it in the roots

Reply:
anyway I removed this check it's redundant, The problem was that the envs source files in the env's package (node_modules) were corrupted so I had to direct the path to the dist